### ISSUE TITLE
New data sources component

### ISSUE THREAD

**Issue by sandhose** [sympa-community/sympa#20]
We want to integrate the new components created for Sympa 7 on the [sympa-vue](https://github.com/sympa-community/sympa-vue) repository in the current version of Sympa.

This is how it might look like in Sympa 7:
![](https://s.sandhose.fr/sympa/ui/flows/add-data-source--mobile.png)

([see here for all the screen, including the desktop ones](https://s.sandhose.fr/sympa/ui/list-admin/data-sources/))

This is how it might look like in the current Sympa UI:
![](https://s.sandhose.fr/sympa/ui/list-admin/data-sources/edit--old.png)

(those are quick sketches and are not pixel-perfect)

This might be as simple as tweaking the tt2 templates, and exposing the lists data sources as a JSON array to the client (we may want to have this interface as an opt-in thing if we don't want to break the current UX). I'm not sure of what's involved in doing that, so I'm calling for help to the backend guys 😶 

The component is currently in development by @ludovicm67 ; see sympa-community/sympa-vue#1

**Comment by racke** [comment]
I'll take a look where the information is kept and how it can be relayed to the new frontend.

**Comment by racke** [comment]
The data sources information is kept in the configuration only as it seems. I will examined it further and check if there is a way to expose it from the 6.2. UI.

**Comment by ikedas** [comment]
There are two types of parameters for inclusion: ``include_*`` (e.g. ``include_sql_query``) and ``*_include`` (e.g. ``member_include``).

The former allows setting of datasource in detail, like presented here[[1]](https://www.sympa.org/manual/parameters-data-sources).  The latter allows restricted setting referring predefined templates[[2]](https://www.sympa.org/manual/list-definition#data_inclusion_file).

On 6.2, the two above share code and configuration schema, however, settings for the former is stored in list config file while the latter is stored in separate *.incl files without UI access.

As an idea, I suggest they would be unified into single framework: separate *.incl files editable by UI.

**Comment by dverdin** [comment]
I strongly agree with  Soji. These two mechanisms are in fine the very same, though they provide different way to create inclusions.

Having separate *.incl files have many advantages which all amount to referential integrity:
- simplicity for mass updates when not using list families (if the SQL host changes for example),
- Smaller set of queries to maintain,
- ability to transfer queries from to site level to virtual hosts, to families (in possible future), to list.

It also extends the delegation properties: you can easily delegate data source management to other users when using *.incl, as all the see are the parameters of a query without seeing the actual query or database host or password, etc.

We should simply improve the current *.incl edition (by giving proper form fields to setup the query parameters) and add an "edit" button that would switch to the actual query parameter edition, such as the the one Sandhose presented.

**Comment by ikedas** [comment]
Code for data source was renewed on #693.  What would be added to realize this FR?
